src/classes/AFFL_MultiRecordType_TEST.cls

```diff
@@ -465,5 +465,39 @@ public with sharing class AFFL_MultiRecordType_TEST {
 
         System.assertEquals(0, allAffiliationList.size()); 
 
-	}	
+	}
+
+    // Test addition of a new affiliation with the same Account doesn't erase the Primary field on Contact
+
+    @isTest
+    public static void newAffiliationInsertTest() {
```

Comment:
This is a great test, that does exactly what we need to. In any other platform this would be perfect. However, in Salesforce unit tests take long(er) to run, which directly affect how long the build takes. I would suggest seeing if one of the existing tests (maybe createPrimaryAffl?) can be modified to also cover this. That way we save build time.